src/Data/Newtype.purs

```diff
@@ -33,41 +34,29 @@ import Data.Semigroup.Last (Last(..))
 -- | unwrap <<< wrap = id
 -- | wrap <<< unwrap = id
 -- | ```
-class Newtype t a | t -> a where
-  wrap :: a -> t
-  unwrap :: t -> a
+class (Coercible t a, Coercible a t) <= Newtype t a | t -> a
```

Comment:
Isn't `Coercible` symmetric? What's the thinking behind giving two `Coercible` superclasses instead of just one and letting the compiler solve the other as needed—or does that not work?

Reply:
Computing subgoals is indeed symmetric, the subgoals of `Coercible a b` are the same than the subgoals of `Coercible b a`, but the compiler doesn’t deduce `Coercible b a` from `Coercible a b`.

This means that we need `Newtype t a` to imply both `Coercible t a` and `Coercible a t` for `unwrap` and `wrap` to compile:

```purs
module Example where

import Safe.Coerce (coerce)
import Prim.Coerce (class Coercible)

class Coercible t a <= Newtype t a | t -> a

wrap :: forall t a. Newtype t a => a -> t
wrap = coerce
```

```
Error found:
in module Example
at Example.purs:9:8 - 9:14 (line 9, column 8 - line 9, column 14)

  No type class instance was found for
                            
    Prim.Coerce.Coercible a0
                          t1
                            

while checking that type forall (a :: Type) (b :: Type). Coercible a b => a -> b
  is at least as general as type a0 -> t1
while checking that expression coerce
  has type a0 -> t1
in value declaration wrap

where t1 is a rigid type variable
        bound at (line 9, column 8 - line 9, column 14)
      a0 is a rigid type variable
        bound at (line 9, column 8 - line 9, column 14)

See https://github.com/purescript/documentation/blob/master/errors/NoInstanceFound.md for more information,
or to contribute content related to this error.
```

This is something we might be able to do though, I’ll look into it.